NitroxPatcher/Patches/Fire_Douse_Patch.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Reflection;
+using Harmony;
+using NitroxClient.MonoBehaviours;
+
+namespace NitroxPatcher.Patches
+{
+    /// <summary>
+    /// Both owners and non-owners will be able to douse fires on a ship.
+    /// </summary>
+    public class Fire_Douse_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(Fire);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("Douse", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(float) }, null);
+
+        public static void Postfix(Fire __instance, float amount)
```

Comment:
It looks like the Fire.Douse() code is called via unity update (every frame).  This appears that it will cause a massive amount of packets to go out.  In the past, we have stored a buffer and broadcasted every 10% or so.  You can see an example in construction packet sending code.

Reply:
Color selection (in the cyclops) faces the same issue. I *tried* to make a generic approach to ratelimiting such packets, but haven't finished it yet. See https://github.com/MarijnS95/Nitrox/tree/ratelimiter.

Reply:
I completely forgot to look into rate limiting this, good thing you caught it. 

I found the relevant buffer code, and it looks like I can use most of it. I think I'm going to modify it to send a packet once the player douses the fire a set amount or extinguishes it since the health of the fire can grow or shrink rapidly.